core/service/src/engine/backup_operator.rs

```diff
@@ -389,6 +406,118 @@ where
+        let cur_op_role = Role::indexed_from_one(cur_recovery_output.operator_role as usize);
+        if cur_op_role != my_role {
+            tracing::warn!(
+                    "Received recovery output for operator role {}, but current server's role is {}. The output will be ignored.",
+                    cur_recovery_output.operator_role,
+                    my_role.one_based()
+                );
+            continue;
+        }
+        if cur_recovery_output.custodian_role == 0
+            || cur_recovery_output.custodian_role > custodian_recovery_outputs.len() as u64
+        {
+            tracing::warn!(
+                    "Received recovery output with invalid custodian role {}. The output will be ignored.",
+                    cur_recovery_output.custodian_role,
+                );
+            continue;
+        }
+        let cur_verf = match recovery_material.custodian_context().custodian_nodes.get(
+            &Role::indexed_from_one(cur_recovery_output.custodian_role as usize),
+        ) {
+            Some(custodian_setup_msg) => &custodian_setup_msg.public_verf_key,
+            None => {
+                tracing::warn!(
+                    "Could not find verification key for custodian role {}",
+                    cur_recovery_output.custodian_role
+                );
+                continue;
+            }
+        };
+        let cur_sig = match k256::ecdsa::Signature::from_slice(&cur_recovery_output.signature) {
+            Ok(sig) => Signature { sig },
+            Err(e) => {
+                tracing::warn!(
+                    "Failed to parse signature for custodian {}: {e}",
+                    cur_recovery_output.custodian_role
+                );
+                continue;
+            }
+        };
+        if internal_verify_sig(
+            &DSEP_BACKUP_CUSTODIAN,
+            &cur_recovery_output.ciphertext,
+            &cur_sig,
+            cur_verf,
+        )
+        .is_err()
+        {
+            tracing::warn!("Could not verify recovery validation material signature");
+            continue;
+        }
+        match <InternalCustodianRecoveryOutput as TryFrom<_>>::try_from(
+            cur_recovery_output.to_owned(),
+        ) {
+            Ok(output) => {
+                if let Some(old_val) = parsed_custodian_rec.insert(output.custodian_role, output) {
+                    tracing::warn!(
+                                "Received multiple recovery outputs for custodian role {}. Only the first one will be used.",
+                                cur_op_role,
+                            );
+                    parsed_custodian_rec.insert(cur_op_role, old_val);
+                }
+            }
+            Err(e) => {
+                tracing::warn!(
+                            "Failed to parse custodian recovery output for operator role {}: {e}. The output will be ignored.",
+                            cur_op_role,
+                        );
```

Comment:
is this the case when the `try_from` fails? I guess that can only happen when custodian or operator role is 0.
We currently don't have this in neither
`client::tests::centralized::custodian_backup_tests::test_decrypt_after_recovery_centralized_negative` nor `client::tests::threshold::custodian_backup_tests::test_decrypt_after_recovery_threshold_negative` and that's why both these tests are failing.